File: C/oop/fib.f90
!!
!> @file fib.f90
!!

!> @brief Compute the Fibonacci serie.
!! @param A Pointer to the array to perform the computation on
!! @param N Array size
SUBROUTINE FIBF90(A,N)

    INTEGER, intent(in) :: N
    REAL(8), dimension(N), intent(out) :: A

    DO I=1,N
     IF (I==1) THEN
        A(I) = 0.0D0
     ELSEIF (I==2) THEN
        A(I) = 1.0D0
     ELSE 
        A(I) = A(I-1) + A(I-2)
     ENDIF
    ENDDO
END

!> @brief Scale an array.
!! @param A Pointer to the array to perform the computation on
!! @param N Array size
subroutine dscal(a, X, N)
   integer, intent(in) :: N
   real(8), dimension(N), intent(out) :: X
   real(8), intent(in) :: a

   X = a*X

end 

File: Python/Cextension/example/fib.c
#include "fib.h"
/**
 *@file fib.c
 */


/**
 * @brief Compute the Fibonacci serie.
 * @param x Pointer to the array to perform the computation on
 * @param N array size
 */
void fib(double *x, int N){

    int i;

    for (i=0; i<N; i++){
        if (i==0){
            *(x+i)=0.0;
        }
        else if (i==1){
            *(x+i) = 1.0;
        }
        else{
            *(x+i) = *(x+i-1) + *(x+i-2);
        }
    }

}